v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/endtoend/ForwardAndReverseMigrationShardedEndToEndIT.java

```diff
@@ -0,0 +1,269 @@
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.testcontainers.shaded.org.apache.commons.lang3.RandomStringUtils;
+
+/**
+ * Integration test for end-to-end Testing of all Spanner migration Flex templates for basic run
+ * including new spanner tables and column rename use-case.
+ */
+@Category({TemplateIntegrationTest.class, SkipDirectRunnerTest.class})
+@TemplateIntegrationTest(DataStreamToSpanner.class)
+@RunWith(JUnit4.class)
+@Ignore("This test is disabled currently")
+public class ForwardAndReverseMigrationShardedEndToEndIT extends EndToEndTestingITBase {
+  private static final String SPANNER_DDL_RESOURCE =
+      "EndToEndTesting/ShardedMigration/spanner-schema.sql";
+
+  private static final String TABLE = "Authors";
+  private static final HashMap<String, String> AUTHOR_TABLE_COLUMNS =
+      new HashMap<>() {
+        {
+          put("id", "INT" + " NOT NULL");
+          put("name", "VARCHAR(200)");
+        }
+      };
+  private static final HashSet<ForwardAndReverseMigrationShardedEndToEndIT> testInstances =
+      new HashSet<>();
+  private static PipelineLauncher.LaunchInfo rrJobInfo;
+  private static PipelineLauncher.LaunchInfo fwdJobInfo;
+  public static SpannerResourceManager spannerResourceManager;
+  private static SpannerResourceManager spannerMetadataResourceManager;
+  private static GcsResourceManager gcsResourceManager;
+  private static PubsubResourceManager pubsubResourceManager;
+
+  private static CloudSqlResourceManager cloudSqlResourceManagerShardA;
+  private static CloudSqlResourceManager cloudSqlResourceManagerShardB;
+
+  private static final Map<String, Object> COLUMNS =
+      new HashMap<>() {
+        {
+          put("name", RandomStringUtils.randomAlphabetic(10));
+        }
+      };
+
+  private static final Integer NUM_EVENTS = 2;
+
+  /**
+   * Setup resource managers and Launch dataflow job once during the execution of this test class.
+   *
+   * @throws IOException
+   */
+  @Before
+  public void setUp() throws IOException, InterruptedException {
+    skipBaseCleanup = true;
+    synchronized (ForwardAndReverseMigrationShardedEndToEndIT.class) {
+      testInstances.add(this);
+      if (rrJobInfo == null || fwdJobInfo == null) {
+        // create Spanner Resources
+        spannerResourceManager =
+            createSpannerDatabase(ForwardAndReverseMigrationShardedEndToEndIT.SPANNER_DDL_RESOURCE);
+        spannerMetadataResourceManager = createSpannerMetadataDatabase();
+
+        // Create MySql Resource
+        cloudSqlResourceManagerShardA =
+            CloudMySQLResourceManager.builder(testName + "ShardA").build();
+        cloudSqlResourceManagerShardB =
+            CloudMySQLResourceManager.builder(testName + "ShardB").build();
+        jdbcSource =
+            createMySqlDatabase(
```

Comment:
Is this method handling mutliple logical shards now? That change was done in previous PR is it?